### PR TITLE
KBV-335 - Address model changes

### PR DESCRIPTION
**Pull request by KaraokeStu**
## Proposed changes
A RFC has been created and agreed by the data and technical architects which affects the way we are modelling address data.

These changes have now been agreed, and we have decided (at the moment) not to use inheritance

### What changed

- Remove the CanonicalAddressWithResidency model and migrate date data into the base model, as per the RFC
- Move UPRN back to numeric data type
- Update CanonicalAddress to contain new ValidFrom and ValidUntil dates
- Update model from ThoroughfareName to StreetName as per Schema
- Ensure dates used are ISO 8691 strings

### Why did it change

The Address model has been in a state of flux for the last few months
This has now been agreed and accepted between development and architecture

### Issue tracking
- [KBV-301](https://govukverify.atlassian.net/browse/KBV-301)
- https://github.com/alphagov/digital-identity-architecture/issues/112
- https://github.com/alphagov/digital-identity-architecture/issues/9
- https://github.com/alphagov/digital-identity-architecture/pull/139

### CODE REVIEW

**Comment by KaraokeStu** [comment]
See https://github.com/alphagov/digital-identity-architecture/pull/139

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/models/CanonicalAddress.java [hunk 1]
```diff
@@ -2,42 +2,56 @@
...
 import uk.gov.di.ipv.cri.address.library.models.ordnancesurvey.Result;
 
+import java.util.Date;
 import java.util.Optional;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @DynamoDBDocument
 public class CanonicalAddress {
-    private String uprn;
+    private Optional<Long> uprn;
```
Comment (jkunle) [inline lib/src/main/java/uk/gov/di/ipv/cri/address/library/models/CanonicalAddress.java:15]: Do you need this to be Optional, you could Optionalise the getter instead

Comment (KaraokeStu) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/models/CanonicalAddress.java]: Done!

File: address.event [hunk 2]
```diff
@@ -13,7 +13,7 @@
...
+  "body": "[\n   {\n      \"uprn\":\"72262801\",\n      \"buildingNumber\": \"8\",\n      \"streetName\": \"GRANGE FIELDS WAY\",\n      \"postTown\": \"LEEDS\",\n      \"postcode\": \"LS10 4QL\",\n      \"countryCode\": \"GBR\",\n      \"validFrom\": \"2010-02-26\",\n      \"validUntil\": \"2021-01-26\"\n   },\n   {\n      \"uprn\": 63094965,\n      \"buildingNumber\": \"15\",\n      \"dependentLocality\": \"LOFTHOUSE\",\n      \"streetName\": \"RIDINGS LANE\",\n      \"postTown\": \"WAKEFIELD\",\n      \"postcode\": \"WF3 3SE\",\n      \"countryCode\": \"GBR\",\n      \"validFrom\": \"2021-01-16\",\n      \"validUntil\": \"2021-08-02\"\n   },\n   {\n      \"uprn\": \"63042351\",\n      \"buildingNumber\": \"5\",\n      \"streetName\": \"GATEWAYS\",\n      \"postTown\": \"WAKEFIELD\",\n      \"postcode\": \"WF1 2LZ\",\n      \"countryCode\": \"GBR\",\n      \"validFrom\": \"2021-08-02\"\n  }\n]",
```
[the remark this answers is not in the document]
Comment (KaraokeStu) [reply address.event]: All now using numeric, rather than string (although both will work)

**Review by jkunle — APPROVED** [review]
👍🏾

**Pull request merged**